pkg/cri/cri.go

```diff
@@ -87,13 +88,56 @@ func initCRIService(ic *plugin.InitContext) (interface{}, error) {
 		return nil, fmt.Errorf("failed to create containerd client: %w", err)
 	}
 
+	// guestPlatform in runtime struct for a runtime helps to support pulling of imager
+	// per runtime class. 
+	// guestPlatform is used to specify an alternate platform to use with platform matcher
+	// so that different versions of the same image can be pulled for different runtime handlers.
+	// guestPlatform.OS and guestPlatform.Architecture are compusory fields to be specified and
+	// if platform is windows, OSVersion needs to be specified too.
+	// Overriding the host's default platform matcher with guestPlatform is not allowed for
+	// windows process isolation as exact OSVersion match between host and guest is required.
+	platformMap := make(map[string]platforms.MatchComparer)
+	for k, ociRuntime := range c.PluginConfig.ContainerdConfig.Runtimes {
+		// consider guesPlatform values only if OS and Architecture are specified
+		if ociRuntime.GuestPlatform.OS != "" &&  ociRuntime.Architecture != "" {
```

Comment:
```suggestion
		if ociRuntime.GuestPlatform.OS != "" &&  ociRuntime.GuestPlatform.Architecture != "" {
```

Reply:
Oops accidentally removed a commit that fixed this while cleaning up the code. Fixed!